packages/slice-machine/src/domain/__tests__/CustomTypeModel.test.ts

```diff
@@ -0,0 +1,190 @@
+    expect(CustomTypeModel.getMainSectionEntry(mockCustomType)).toEqual([
+      "MainSection",
+      MainSection,
+    ]);
+  });
+
+  it("getMainSectionEntry should return undefined if there is are sections", () => {
+    expect(
+      CustomTypeModel.getMainSectionEntry({
+        ...mockCustomType,
+        json: {},
+      })
+    ).toEqual(undefined);
+  });
+
+  it("getSection should return the section matching the key", () => {
+    expect(
+      CustomTypeModel.getSection(mockCustomType, "AnotherSection")
+    ).toEqual(AnotherSection);
+  });
+
+  it("getSection should return undefined if there are no sections", () => {
+    expect(
+      CustomTypeModel.getSection(
+        {
+          ...mockCustomType,
+          json: {},
+        },
+        "MainSection"
+      )
+    ).toEqual(undefined);
+  });
+
+  it("getSectionSliceZoneConfig should return the config of the given section", () => {
+    expect(
+      CustomTypeModel.getSectionSliceZoneConfig(mockCustomType, "MainSection")
+    ).toEqual({
+      choices: {
+        hero_banner: {
+          type: "SharedSlice",
+        },
+        promo_section_image_tiles: {
+          type: "SharedSlice",
+        },
+      },
+    });
+  });
+
+  it("getSectionSliceZoneConfig should return undefined if there are no sections", () => {
+    expect(
+      CustomTypeModel.getSectionSliceZoneConfig(
+        {
+          ...mockCustomType,
+          json: {},
+        },
+        "MainSection"
+      )
+    ).toEqual(undefined);
+  });
+
+  it("createSectionSliceZone should return the given custom type with a slice zone for given section", () => {
+    expect(
+      CustomTypeModel.createSectionSliceZone(mockCustomType, "AnotherSection")
+    ).toEqual({
+      ...mockCustomType,
+      json: {
+        ...mockCustomType.json,
+        AnotherSection: {
+          ...mockCustomType.json.AnotherSection,
+          slices: {
+            type: "Slices",
+            fieldset: "Slice Zone",
+            config: {
+              choices: {},
+            },
+          },
+        },
+      },
+    });
+  });
+
+  it("createSectionSliceZone should return the same custom type if slice zone already exist for given section", () => {
+    expect(
+      CustomTypeModel.createSectionSliceZone(mockCustomType, "MainSection")
+    ).toEqual(mockCustomType);
+  });
+
+  it("convertToPageType should convert the given custom type", () => {
+    expect(CustomTypeModel.convertToPageType(mockCustomType)).toEqual({
+      ...mockCustomType,
+      format: "page",
+    });
+  });
+
+  it("convertToPageType should convert the given custom type with a slice zone for Main section when it doesn't exist", () => {
+    expect(
+      CustomTypeModel.convertToPageType({
+        ...mockCustomType,
+        json: {
+          MainSection: {},
+          AnotherSection,
+        },
+      })
+    ).toEqual({
+      ...mockCustomType,
+      json: {
+        MainSection: {
+          slices: {
```

Comment:
@hypervillain 

Here on the old code, the key was created with a unique name depending on all slice zone of all tabs.

Do we really want that? I don't get it, since each slice zone is specific to a tab.

Old code for the key: `slices${i !== 0 ? i.toString() : ""}`

Reply:
Done, it's required